dataset/data_array.cpp

```diff
@@ -90,9 +90,13 @@ DataArray DataArray::slice(const Slice &s) const {
           m_attrs->slice(s).merge_from(out_attrs), m_name};
 }
 
-DataArray &DataArray::setSlice(const Slice &s, const DataArray &array) {
+void DataArray::validateSlice(const Slice &s, const DataArray &array) {
   expect::coordsAreSuperset(slice(s), array);
-  // TODO Need dry-run mechanism for mask handling?
+  masks().validateSlice(s, array.masks());
```

Comment:
Do we also need to check that `setSlice` on the data could not fail? Is suppose the coord check is catching a lot of potential problems, but not all, e.g., if only one arg has variances, or if units differs? If these aspects are not checked there are two problems:

- Masks get modified.
- If this is applied to dataset items, other dataset items may have beed modified already.

Reply:
I suppose basically we also need `Variable::validateSlice`?